frontend/components/Cell.js

```diff
@@ -131,7 +127,36 @@ export const Cell = ({
     const set_waiting_to_run_smart = (x) => set_waiting_to_run(x && should_set_waiting_to_run_ref.current)
 
     const cell_api_ready = useCellApi(node_ref, published_object_keys, pluto_actions)
-
+    const on_delete = useCallback(() => {
+        pluto_actions.confirm_delete_multiple_from_cell("Delete", cell_id, selected)
+    }, [pluto_actions, selected, cell_id])
+    const on_submit = useCallback(() => {
+        if (!disable_input_ref.current) {
+            set_waiting_to_run_smart(true)
+            pluto_actions.set_and_run_multiple([cell_id])
+        }
+    }, [pluto_actions, set_waiting_to_run, cell_id])
+    const on_change_cell_input = useCallback(
+        (new_code) => {
+            if (!disable_input_ref.current) {
+                if (code_folded && cm_forced_focus != null) {
+                    pluto_actions.fold_remote_cell(cell_id, false)
+                }
+                on_change(new_code)
+            }
+        },
+        [code_folded, cm_forced_focus, pluto_actions, on_change]
+    )
+    const on_add_after = useCallback(() => {
+        pluto_actions.add_remote_cell(cell_id, "after")
+    }, [pluto_actions, cell_id, selected])
+    const on_code_fold = useCallback(() => {
```

Comment:
`on_code_fold` and `on_run` are refactored to use new actions that pick the state immediately, instead of passing that down to props